codec/encode.go

```diff
@@ -514,6 +514,19 @@ func (f *encFnInfo) kStruct(rv reflect.Value) {
 	toMap := !(fti.toArray || e.h.StructToArray)
 	newlen := len(fti.sfi)
 
+	var unknownFields map[string]interface{}
+	// TODO: Merge unknown fields with known ones rather than just
+	// tack the unknown ones at the end.
+	if fti.ufh {
+		if ufh, proceed := f.getValueForMarshalInterface(rv, fti.ufhIndir); proceed {
+			if !toMap {
```

Comment:
A fully general solution would also handle the `!toMap` case, as well as define what happens when there's an unknownField and a known field with the same name.